core/dbt/contracts/graph/parsed.py

```diff
@@ -258,10 +262,16 @@ def _deserialize(cls, dct: Dict[str, int]):
             return cls.from_dict(dct)
 
     def _persist_column_docs(self) -> bool:
-        return bool(self.config.persist_docs.get('columns'))
+        if hasattr(self.config, 'persist_docs'):
+            assert isinstance(self.config, NodeConfig)
```

Comment:
I get that this is a mypy trick, but if this asserts false, it'll raise to our users right? Might want to put a user-friendly exception before the assert.

Reply:
The 'assert' is for situations where we will never encounter an actual error, it is only to make mypy happy. So there is no need for a user friendly exception.

Reply:
I just don't see any mechanical proof that we could never reach this state. I might be being overly cautious, but code that throws makes me nervous.